sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/reflection/ReflectionUtils.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl.reflection
+
+import java.lang.reflect.Method
+import java.util
+
+object ReflectionUtils {
+
+  private implicit val stringArrayOrdering: Ordering[Array[String]] =
+    Ordering.fromLessThan(util.Arrays.compare[String](_, _) < 0)
+  implicit val methodOrdering =
+    Ordering.by((m: Method) => (m.getName, m.getReturnType.getName, m.getParameterTypes.map(_.getName)))
```

Comment:
We are still not protecting against nodes with different versions during a rolling upgrade.

Reply:
At least now this won't produce a broken cluster that stays broken once the roll completed.